[refactor] 채팅방 목록 조회 기능 수정

pie2457 opened the pull request:
## Issues
- #68 

## What is this PR? 👓
채팅방 목록 조회 시 데이터를 잘못 가져와 코드 수정에 대한 PR입니다.

## Key changes 🔑
- `chatroomId`가 아닌 `chatLogId`를 가져오는것을 수정
- 채팅하는 상대방에 대한 정보를 가져오지 못하는 것을 수정
- `chatroomId`에 해당하는 `chatLog`의 마지막 메세지와 그 시간을 가져오지 못하는 것을 수정

pie2457 committed:
refactor #68 : 채팅방 목록을 불러올 때 데이터를 잘못가져와 코드 수정

- ChatLogRepository : chatLog 테이블에서 chatroomId에 해당하는 가장 마지막 메세지와 메세지의 시간을 가져옴
- ChatPaginationRepository : queryDsl로 response 데이터를 한번에 처리하지 않고 필요한 정보만 가져오게 함.
- ChatroomListResponse : queryDsl로 못가져온 데이터를 가져다가 채워넣어줌
- ChatroomService : response안에 들어갈 데이터들을 찾아오게끔 수정

pie2457 committed:
refactor #68 : 테스트 코드 수정

23Yong commented on src/main/java/pie/tomato/tomatomarket/application/chat/ChatroomService.java:
```diff
@@ -38,15 +41,30 @@ public CustomSlice<ChatroomListResponse> findAll(Principal principal, int size,
 	}
 
+	private void assignResponseInfo(Principal principal, ChatroomListResponse chatroomListResponse) {
+		ChatLog findChatLog = chatLogRepository.findFirstByChatroomId(chatroomListResponse.getChatroomId());
```
findChatLog 하면 뭔가 `채팅 메시지를 찾는다` 라는 느낌이 드는데
acquiredChatLog 나 obtainedChatLog 혹은 그냥 chatLog 라 네이밍해도 괜찮을 것 같아요!

pie2457 replied:
넵 좋은거같아요 !! 감사합니다ㅏ :)

pie2457 replied:
넹 그렇게 수정할게요 감사합니다 !

23Yong commented on src/main/java/pie/tomato/tomatomarket/presentation/chat/response/ChatroomListResponse.java:
```diff
@@ -18,4 +21,19 @@ public class ChatroomListResponse {
+	}
+
+	public void assignPartnerInfo(Chatroom chatroom, Principal principal) {
+		if (chatroom.getSeller().getId() == principal.getMemberId()) {
```
chatroom.isSeller(prinicpal.getMemberId()) 로 하면 훨씬 직관적일 것 같아요!

pie2457 replied:
넹 좋네여 !! 그렇게 수정할게요 감삼다 ~~~!

23Yong commented on src/main/java/pie/tomato/tomatomarket/infrastructure/persistence/chat/ChatLogRepository.java:
```diff
@@ -18,6 +18,10 @@ public interface ChatLogRepository extends JpaRepository<ChatLog, Long> {
 		+ "where chatLog.chatroom.id = :chatroomId and chatLog.id >= :messageIndex")
 	void changeToReadState(@Param("chatroomId") Long chatroomId, @Param("messageIndex") Long messageIndex);
 
+	@Query(value = "SELECT * FROM chat_log chatLog "
+		+ "WHERE chatLog.chatroom_id = :chatroomId ORDER BY chatLog.id DESC LIMIT 1", nativeQuery = true)
+	ChatLog findFirstByChatroomId(@Param("chatroomId") Long chatroomId);
```
`findTopByChatRoomIdOrderByChatRoomIdDesc` 이렇게 하면 위에 `@Query` 를 작성할 필요가 없을 것 같아요!

https://www.baeldung.com/spring-data-sorting

pie2457 replied:
반환 타입을 컬렉션으로 받지 않고 엔티티로 받으면 한개만 나올 수 있는가 궁금하네요 !! 수정해보겠습니다 감사해용!!

23Yong replied:
https://www.baeldung.com/spring-data-jpa-findfirst-vs-findtop

findTop 을 쓰면 엔티티 하나만 반환하네용

23Yong reviewed (approved):
수고하셨습니다!

pie2457 committed:
refactor #68 : 리뷰사항 반영

- ChatroomService : 변수명 변경
findChatLog -> acquiredChatLog
- ChatLogRepository : 메서드명을 변경하고 @Query 삭제
- Chatroom, ChatroomListResponse : isSeller 메서드 추
